File: iSOLID.RiskAdvisory.App/wwwroot/js/models/dataExceptionCategory.js
leala.models.dataExceptionCategory = function (d) {

    var self = this;

    self.name = ko.observable().extend({ required: true, maxLength: 100 });
    self.code = ko.observable().extend({ required: true, maxLength: 10 });
    self.exceptions = ko.observableArray();

    if (d) {
        self.name(d.name);
        self.code(d.code);

        self.exceptions.removeAll();

        var exceptions = new Array();
        for (var i = 0; i < d.exceptions.length; i++) {
            var exception = new leala.models.dataException(d.exceptions[i]);
            exceptions.push(exception)
        }
        self.exceptions(exceptions);
    }

    self.displayText = ko.pureComputed(function () {
        if (self.name() && self.code()) {
            return self.code() + ': ' + self.name();
        }
        return null;
    });
};

File: iSOLID.RiskAdvisory.Domain/Model/DataExceptionCategory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace iSOLID.RiskAdvisory.Domain.Model
{
    public class DataExceptionCategory : Entity
    {
        public DataExceptionCategory()
        {
            this.Exceptions = new List<DataException>();
        }

        public List<DataException> Exceptions { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        [Required]
        [MaxLength(10)]
        public string Code { get; set; }
    }
}
